bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -75,11 +75,20 @@ type KubeadmConfigSpec struct {
 	// +optional
 	Mounts []MountPoints `json:"mounts,omitempty"`
 
-	// preKubeadmCommands specifies extra commands to run before kubeadm runs
+	// bootCommands specifies extra commands to run very early in the boot process via the cloud-init bootcmd
+	// module. This is typically run in the cloud-init.service systemd unit. This has no effect in Ignition.
```

Comment:
If there is no equivalent for the ignition format, it is good to prevent user confusion by introducing a validation webhook check on `kubeadmConfig.spec.format == “cloud-init”` for this field to be set.

Reply:
`bootstrap/kubeadm/internal/webhooks/kubeadmconfig.go` seems to be the place for validation webhooks code

Reply:
Do you mean to validate that `format == "clout-init"` only if ignition is not used? Is that based on `feature.KubeadmBootstrapFormatIgnition` or is there another way of checking if ignition should be used?

Reply:
Format field on the config designates the type of configuration used. If it is cloud-init, then it is not ignition :)

Reply:
Okay thanks, is this how you meant? https://github.com/kubernetes-sigs/cluster-api/pull/11271/commits/b7c575a63f5f0aafbb6108076859ac36db7d6d8e

Also, I believe the [default value for the format](https://github.com/kubernetes-sigs/cluster-api/blob/6fb468d53a18804bc2b585138a12a68f008efa7e/bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go#L35) is "cloud-config", but please let me know if I misunderstood and it should be "cloud-init".